server/service/src/invoice/outbound_return/generate_outbound_return_lines.rs

```diff
@@ -0,0 +1,148 @@
+use crate::{service_provider::ServiceContext, ListError, ListResult};
+use repository::{EqualFilter, StockLine, StockLineFilter, StockLineRepository};
+use util::uuid::uuid;
+
+#[derive(Debug, Clone)]
+pub struct OutboundReturnLine {
+    pub id: String,
+    pub reason_id: Option<String>,
+    pub comment: Option<String>,
+    pub number_of_packs: u32,
+    pub stock_line: StockLine,
+}
+
+pub fn generate_outbound_return_lines(
+    ctx: &ServiceContext,
+    store_id: &str,
+    stock_line_ids: Vec<String>,
+    item_id: Option<String>,
+    return_id: Option<String>,
+) -> Result<ListResult<OutboundReturnLine>, ListError> {
+    if stock_line_ids.is_empty() && item_id.is_none() && return_id.is_none() {
+        return Ok(ListResult {
+            count: 0,
+            rows: vec![],
+        });
+    }
```

Comment:
Should this instead be a validation step? Throw an error if the query inputs are empty?

Or alternatively, is there a way that I could define the filter below so that if nothing is passed in, the query returns no lines, as opposed to all the available stock lines?